apps/wakunode2/external_config.nim

```diff
@@ -225,9 +225,13 @@ type
       defaultValue: false
       name: "keep-alive" }: bool
 
+    topics* {.
```

Comment:
No action required, just a thought.

afaik its agreed to remove `topics` (which is pubsub topic) in favour of `content-topic`, since we want to abstract the user away from that.

so perhaps this change is not neccesary? i mean, why renaming `topics` to `pubsub-topic` if we are removing in it soon?

Reply:
yup, we are aligned on the first part.

> None of this, however, necessarily determines which shards/pubsub topic a node is configured by default to relay on. I might want to support all public waku network shards without even knowing a single content topic, so would continue needing a pubsubTopic config item for that.

If you want to to support all shard, perhaps what we need is as `all-shards` flag. This issue https://github.com/waku-org/nwaku/issues/1892 might also show my intentions.

But yeah, by now not against leaving the pubsub flag there.

Reply:
For now at least, I think the light protocols should run the same "middleware" and populate the pubsub topic in e.g. the filter subscribe request before sending it. This may only be a first phase, but allows the application to gain from the abstraction of not having to use pubsub topics on the API, while remaining "safe" by not violating any existing protocols or having implicit assumption that the the full node is on the same version/generation of autosharding.

Reply:
resolving

happy to keep the pusub topic by now, but in the future i dont think it makes sense. i get the "remaining safe" due to generation, but this is something that i think we have to discuss. exposing the app/users to this pubsub topic + generation concept adds more load and moving parts that should be abstracted away.

in other words, it solves the problem from a technical perspective but paying in complexity for the app (developers building on top)